src/foam/nanos/script/Script.js

```diff
@@ -148,6 +157,32 @@ foam.CLASS({
         ps.flush();
         setOutput(baos.toString());
     `
+    },
+    {
+      name: 'poll',
```

Comment:
This implementation is fine, but you could have also made it a listener and set mergeDelay: 2000, and then call yourself recursively, which would have been less code.